src/containers/Elements/NodeProvider.js

```diff
@@ -95,20 +124,28 @@ function makeMapStateToProps() {
   const getProviderNodes = makeGetProviderNodes();
 
   return function mapStateToProps(state, props) {
-    const interaction = (props.selectable && 'selectable') || (props.draggable && 'draggable') || 'none';
+    const currentIds = { promptId: props.prompt.id, stageId: props.stage.id };
+    const isOrigin = isEqual(state.draggable.draggingFromIds, currentIds);
```

Comment:
Um, I'll have to look at the `hit` calculator to answer that. I'm not sure I understand how that works fully.

Reply:
It loops through the registered areas and returns a list of matches, I think there's a little bit of meta data provided so that the draggable can decide what to do with the event, it might be possible to add this origin information there, but I'm not sure.

Reply:
I'll take a closer look at it then and see what I can learn.